aten/src/ATen/native/LossNLL.cpp

```diff
@@ -51,6 +53,56 @@ TORCH_META_FUNC(nll_loss_forward)
 
   set_output(1, {}, self.options());
 }
+
+TORCH_META_FUNC(nll_loss_backward)
+(const Tensor& grad_output,
+ const Tensor& self,
+ const Tensor& target,
+ OptionalTensorRef weight_opt,
+ int64_t reduction,
+ int64_t ignore_index,
+ const Tensor& total_weight) {
+  TORCH_CHECK(
+      self.dim() > 0 && self.dim() <= 2, "input tensor should be 1D or 2D");
+  TORCH_CHECK(
+      target.dim() == 1,
+      "1D target tensor expected, multi-target not supported");
+  TORCH_CHECK(
+      self.size(0) == target.size(0),
+      "size mismatch (got input: ",
+      self.sizes(),
+      ", target: ",
+      target.sizes(),
+      ")")
+  TORCH_CHECK(
+      total_weight.numel() == 1,
+      "expected total_weight to be a  single element tensor, got: ",
+      total_weight.sizes(),
+      " (",
+      total_weight.numel(),
+      " elements)");
+
+  auto weight = weight_opt.getTensorRef();
+
+  TORCH_CHECK(
+      !weight.defined() || weight.numel() == self.size(-1),
+      "weight tensor should be defined either for all or no classes");
+
+  const auto n_dims = self.dim();
+
+  if (reduction == Reduction::None && n_dims == 2) {
+    const auto batch_size = self.size(0);
+    check_dim_size(grad_output, 1, 0, batch_size);
+  } else {
+    TORCH_CHECK(
+        grad_output.dim() <= 1 && grad_output.numel() == 1,
+        "Expected a single element grad_output tensor, but got: ",
+        grad_output.sizes());
+  }
+
+  TensorOptions options(self.options());
```

Comment:
TensorOptions is entirely functional, so you can do this fluently (`self.options().memory_format(...)`)